src/service/loader.js

```diff
@@ -0,0 +1,66 @@
+import log from "loglevel";
+import { Agent } from "https";
+import { readFileSync } from "fs";
+
+export default class setup {
+  constructor(dotenv) {
+    let logLevel = process.env.LOG_LEVEL || "DEBUG";
+    log.setLevel(log.levels[logLevel]);
+
+    const result = dotenv.config();
+    if (result.error) {
+      throw result.error;
+    }
+    console.log("Environment variables loaded.");
+
+    this.url = process.env.MESH_URL || "https://localhost:8700";
+    this.sharedKey = process.env.MESH_SHARED_KEY || "TestKey";
+    this.sandbox = process.env.MESH_SANDBOX || "true";
+    if (this.sandbox === true) {
+      console.log("Running in sandbox mode");
+      // just setup to ignore self-signed certs
+      senderAgent = new Agent({
+        rejectUnauthorized: false,
+      });
+
+      receiverAgent = new Agent({
+        rejectUnauthorized: false,
+      });
+    } else {
+      this.senderAgent = new Agent({
+        cert: readFileSync(process.env.MESH_SENDER_CERT_LOCATION),
+        key: readFileSync(process.env.MESH_SENDER_KEY_LOCATION),
+        rejectUnauthorized: false,
+      });
+      this.receiverAgent = new Agent({
+        cert: readFileSync(process.env.MESH_RECEIVER_CERT_LOCATION),
+        key: readFileSync(process.env.MESH_RECEIVER_KEY_LOCATION),
+        rejectUnauthorized: false,
+      });
+    };
+    this.senderMailboxID = process.env.MESH_SENDER_MAILBOX_ID || "X26ABC1";
```

Comment:
X26ABC1 is this meaningful in any way? If yes we shouldn't be hardcoding. If no, would be more helpful to shoud that its a placeholders i.e "A00AAA0"

Reply:
These are ids Andrew is using

Reply:
the hardcoded Ids run when sandbox is toggled

Reply:
Zain is concerned that mail box id being exposed in a public repo is not a good idea, and I agree.

I think maybe the best approach is if a mail box id is not setup in the environment variable an error should be thrown/returned rather than using a hardcoded sandbox mail box id.

Reply:
They are already visible in index.js within the repo, I can remove them in my files, would you also want them removed from the entire repo?

Reply:
The ID’s are for the defaults that are created when you generate a local sandbox, those values were gathered from another NHS git repo so anyone that wants to know them can just go a look at the mesh-sandbox github repo